### ISSUE TITLE
TypeLoadException. Can't load non-generic TableQuerySegment in .NET standard 8.1

### ISSUE THREAD

**Issue by Crokus** [Azure/azure-storage-net#416]
I'm developing .NET Standard 1.3 library which has dependency on WindowsAzure.Storage 8.1 package.
I've created NuGet package from my library and hosted it on my local feed.

Then I've created a new .NET 4.6 console app where I wanted to test my library. After installing my package everything compiled correctly but since my library uses `ExecuteQuerySegmentedAsync` with non-generic `TableQuerySegment` after running the app I've got a TypeLoadException:

> Could not load type 'Microsoft.WindowsAzure.Storage.Table.TableQuerySegment' from assembly 'Microsoft.WindowsAzure.Storage, Version=8.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.

As I can see there is only generic`TableQuerySegment<T>` class in `/net45` assembly.

How can I make the .NET standard lib that would target both .NET Standard 4.6 and .NET core using  WindowsAzure.Storage 8.1?

**Comment by erezvani1529** [comment]
Hi @Crokus,

Apologies for the delay in response. As I have mentioned on other issues, we are currently maintaining 2 code bases for our desktop and NetStandard support respectively.

The support for POCO entities was not initially added to the RT/NetCore code-base which is the reason why we did not support the generic TableQuerySegment. However, now that the generic versions are supported we recommend that you move towards using them in order to resolve your issue. Is there a reason that the generic version (which is supported in both code-bases), does not work for you?

Thanks,
Elham

**Comment by Crokus** [comment]
Hi @erezvani1529

Root of my issue was that I was using non-generic version and I've overlooked that there're different overloads of `ExecuteQuerySegmentedAsync ` in the two code-bases.

In desktop there's a overload that takes non-generic `TableQuery` and returns `Task<TableQuerySegment<DynamicTableEntity>` but in NetCore there's no such overload.

Once I've examined all possible overloads supported in NetCore I've changed to code to use non-generic overload that takes 'TableQuery<T>' and returns `TableQuerySegment<T>`.

Now it works. Thanks for your help and guiding to the non-generic version of the API.
